### PR TITLE
/issues/121 Change output location of dll

### PR DESCRIPTION
**Pull request by sanderlanghorstredhotminute**

### CODE REVIEW

**Comment by dawoe** [comment]
Is fix for #121

File: Umbraco.ModelsBuilder/Configuration/Config.cs [hunk 1]
```diff
@@ -63,6 +64,10 @@ private Config()
             ModelsDirectory = HostingEnvironment.IsHosted
                 ? HostingEnvironment.MapPath(DefaultModelsDirectory)
                 : DefaultModelsDirectory.TrimStart("~/");
+            BinDirectory = HostingEnvironment.IsHosted
+                ? HostingEnvironment.MapPath(DefaultBinDirectory)
+                : DefaultModelsDirectory.TrimStart("~/");
```
Comment (zpqrtbnk) [inline Umbraco.ModelsBuilder/Configuration/Config.cs]: Shouldn't this be `DefaultBinDirectory.TrimStart("~/")` ?

Comment (sanderlanghorstredhotminute) [reply Umbraco.ModelsBuilder/Configuration/Config.cs]: It should! This is an error by copying the ModelsDirectory.

**Comment by sanderlanghorstredhotminute** [comment]
Note that additional rights may be needed to create folders and files outside the bin or app_data folder

**Comment by zpqrtbnk** [comment]
oh - i've made a change to the code and might have break everything maybe - can you check? idea is this: the folder is required to be within the web root, for security reasons - same as with app_data - but for app_data we have a special "accept unsafe" setting - haven't created such a setting for bin, should we?

File: Umbraco.ModelsBuilder/Configuration/Config.cs [hunk 2]
```diff
@@ -165,6 +170,27 @@ private Config()
...
+                    ? HostingEnvironment.MapPath("~/")
+                    : Directory.GetCurrentDirectory();
+                if (root == null)
+                    throw new ConfigurationErrorsException("Could not determine root directory.");
+
+                var path = HostingEnvironment.IsHosted
+                    ? HostingEnvironment.MapPath(value)
+                    : Path.GetFullPath(value);
+
+                if (path != null && path.StartsWith(root))
```
Comment (sanderlanghorstredhotminute) [inline Umbraco.ModelsBuilder/Configuration/Config.cs]: @zpqrtbnk this part checks if the path is within the root. Though additional rights may be needed if the application's rights are restricted.

**Pull request merged**